src/main/controllers/features/common/amlBodyMembershipNumberController.ts

```diff
@@ -2,38 +2,67 @@ import { NextFunction, Request, Response } from "express";
+import { saveDataInSession } from "../../../common/__utils/sessionHelper";
 
 export const get = async (req: Request, res: Response, next: NextFunction) => {
     const lang = selectLang(req.query.lang);
     const locales = getLocalesService();
     const session: Session = req.session as any as Session;
-    const acspData: ACSPData = session?.getExtraData(USER_DATA)!;
-    const acspType: string = acspData?.typeOfBusiness!;
-    const selectedAMLSupervisoryBodies: string[] = session?.getExtraData(AML_SUPERVISOR_SELECTED);
-    const previousPage: string = getPreviousPage(acspType);
+    const currentUrl: string = BASE_URL + AML_MEMBERSHIP_NUMBER;
+
+    try {
+        // get data from mongo and save to session
+        const acspData = await getAcspRegistration(session, session.getExtraData(SUBMISSION_ID)!, res.locals.userId);
+        saveDataInSession(req, USER_DATA, acspData);
 
-    res.render(config.AML_MEMBERSHIP_NUMBER, {
-        ...getLocaleInfo(locales, lang),
-        previousPage: addLangToUrl(previousPage, lang),
-        currentUrl: BASE_URL + AML_MEMBERSHIP_NUMBER,
-        selectedAMLSupervisoryBodies
+        const acspType: string = acspData?.typeOfBusiness!;
+        const previousPage: string = getPreviousPage(acspType);
 
-    });
+        // collect selected AMLs
+        const selectedAMLSupervisoryBodies: string[] = [];
+        const amlSupervisoryBody = new AmlSupervisoryBodyService();
+        amlSupervisoryBody.getSelectedAML(acspData, selectedAMLSupervisoryBodies);
+
+        // collect membership numbers to render the page with saved data
+        let payload;
+        amlSupervisoryBody.getMembershipNumbers(acspData, selectedAMLSupervisoryBodies, payload);
+
+        res.render(config.AML_MEMBERSHIP_NUMBER, {
+            ...getLocaleInfo(locales, lang),
+            previousPage: addLangToUrl(previousPage, lang),
+            currentUrl,
+            selectedAMLSupervisoryBodies,
+            payload
+        });
+    } catch (err) {
+        logger.error(GET_ACSP_REGISTRATION_DETAILS_ERROR);
+        const error = new ErrorService();
+        error.renderErrorPage(res, locales, lang, currentUrl);
+    }
 };
 
 export const post = async (req: Request, res: Response, next: NextFunction) => {
     const lang = selectLang(req.query.lang);
     const locales = getLocalesService();
     const session: Session = req.session as any as Session;
-    const acspData: ACSPData = session?.getExtraData(USER_DATA)!;
-    const selectedAMLSupervisoryBodies: string[] = session?.getExtraData(AML_SUPERVISOR_SELECTED);
+    const acspData: AcspData = session?.getExtraData(USER_DATA)!;
     const acspType: string = acspData?.typeOfBusiness!;
     const previousPage: string = getPreviousPage(acspType);
+    const currentUrl: string = BASE_URL + AML_MEMBERSHIP_NUMBER;
+
+    // collect selected AMLs
+    const selectedAMLSupervisoryBodies: string[] = [];
+    const amlSupervisoryBody = new AmlSupervisoryBodyService();
```

Comment:
I think this can be decalred and defined once and used in both the methods.

Reply:
Addressed